feat: show new way to register (#1227)
WalletConnect/walletconnect-docs: docs/api/notify/migration-guide.mdx

import Container from '../../components/Container'
import Tabs from '@theme/Tabs'
import TabItem from '@theme/TabItem'
import PlatformTabs from '../../components/PlatformTabs'
import PlatformTabItem from '../../components/PlatformTabItem'
import CloudBanner from '../../components/CloudBanner'

# Migration Guide

<PlatformTabs
  groupId="api"
  activeOptions={["web"]}
>

<PlatformTabItem value="web">

## Migrating from 0.14.x to 0.15.x 

> There is a breaking change with registering between the minor versions.
> - Instead of passing an `onSign` function, signing must be done separately
> - `isLimited` has been removed in favor of `allApps`, its inverse. Wherever `isLimited` was true, `allApps` would be false.
> - Instead of calling register immediately, one must first prepare a registration via `prepareRegistration`
> - After getting the `registerParams` and a `message`, the message must be signed and the signature passed to `register` along with `registerParams`.

Reviewer: Some copy tweaks:

```suggestion
There is a breaking change with registering between the minor versions.

- `isLimited` has been replaced with `allApps`, its inverse, to avoid double-negatives. Whenever `isLimited` is set to false, `allApps` should now be true and vise-versa.
- Instead of passing an `onSign` callback to `register()`, registration takes place in multiple steps: `prepareRegistration()`, the signature, and `register()`
  - Instead of calling `register()` immediately, one must first prepare a registration via `prepareRegistration()`
  - After getting the `registerParams` and a `message`, the message must be signed and the signature passed to `register()` along with `registerParams`.
```

### Old
```javascript
const account = `eip155:1:0x63Be2c680685d2A9620c11b0068291261aa62d76`
const onSign = (message: string) => ethersWallet.signMessage(message)

await notifyClient.register({
  account,
  onSign,
  domain: 'com.example.app.rn', // pass your app's bundle identifier.
  isLimited: false 
})
```

### New Equivalent
```javascript
const account = `eip155:1:0x63Be2c680685d2A9620c11b0068291261aa62d76`
const domain =  'app.mydomain.com', // pass the domain (i.e. the hostname) where your dapp is hosted.

> const allApps =  false 

Reviewer: ```suggestion
const allApps = true
```

Author: keeping it as `true` since the old example has `isLimited = false`.

Reviewer: The migration is incorrect though, `isLimited: false` should change to `allApps: true` not `allApps: false`

// No need to register and sign message if already registered.
if (notifyClient.isRegistered({ account, domain, allApps })) return;

> const {registerParams, message}  = notifyClient.prepareRegistration({

Reviewer: ```suggestion
const { registerParams, message }  = notifyClient.prepareRegistration({
```
